core/src/main/java/io/questdb/griffin/SqlKeywords.java

```diff
@@ -62,6 +79,31 @@ public static boolean isAlterKeyword(CharSequence tok) {
                 && (tok.charAt(i) | 32) == 'r';
     }
 
+    public static boolean isFirstFunction(CharSequence tok) {
+        if (tok.length() != 5) {
+            return false;
+        }
+
+        int i = 0;
+        return (tok.charAt(i++) | 32) == 'f'
+                && (tok.charAt(i++) | 32) == 'i'
+                && (tok.charAt(i++) | 32) == 'r'
+                && (tok.charAt(i++) | 32) == 's'
+                && (tok.charAt(i) | 32) == 't';
+    }
+
+    public static boolean isLastFunction(CharSequence tok) {
+        if (tok.length() != 4) {
+            return false;
+        }
+
+        int i = 0;
+        return (tok.charAt(i++) | 32) == 'l'
+                && (tok.charAt(i++) | 32) == 'a'
+                && (tok.charAt(i++) | 32) == 's'
+                && (tok.charAt(i) | 32) == 't';
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 96:104, 260:268, 272:280, 380:388, 435:444, 448:456, 476:484, 540:548, 565:573, 727:735, 770:778, 809:817, 821:829, 908:916, 963:971, 1001:1009, 1304:1312, 1362:1370, 1428:1436, 1454:1462.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.